sunpy/database/database.py

```diff
@@ -354,20 +363,26 @@ def download(self, *query, **kwargs):
         """
         if not query:
             raise TypeError('at least one attribute required')
-        client = kwargs.pop('client', None)
-        path = kwargs.pop('path', None)
-        progress = kwargs.pop('progress', False)
-        if kwargs:
-            k, v = kwargs.popitem()
+        
+        kwargs_copy = kwargs.copy()
```

Comment:
but if it's empty... then the kwarg would have the same elements that you took away, would not `kwarg.get` would be better for this? and then check whether the `kwargs` have different values than the one you want to pass?
I think I would find that way more readable than this.